File: src/ruby/Ecosistema/Biotopo.rb
require('./Ecosistema/Locacion.rb');
require('./Ecosistema/Biosenosis.rb');

module Ecosistema
	public
	class Biotopo

		# @ReturnType String
		public
		def getNombre()
			return @_nombre
		end

		# @ParamType aNombre String
		# @ReturnType void
		public
		def setNombre(aNombre)
			@_nombre = aNombre
		end

		# @ReturnType ecosistema.Locacion
		public
		def getLocation()
			return @_location
		end

		# @ParamType aLocation ecosistema.Locacion
		# @ReturnType void
		public
		def setLocation(aLocation)
			@_location = aLocation
		end

		public
		def initialize()
			# @AttributeType String
			@_nombre
			# @AttributeType ecosistema.Locacion
			@_location
			# @AttributeType ecosistema.Biosenosis
			# # @AssociationType ecosistema.Biosenosis
			@_atLocation_location___Location_
			# @AttributeType ecosistema.Locacion
			# # @AssociationType ecosistema.Locacion
			# # @AssociationMultiplicity 1
			# # @AssociationKind Aggregation
			@_laLocacion

			# Not yet implemented
		end
	end
end

File: src/python/ecosistema/Biotopo.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
from ecosistema import Locacion
from ecosistema import Biosenosis
from typing import List

class Biotopo(object):
	def getNombre(self) -> String:
		return self.___nombre

	def setNombre(self, aNombre : String) -> None:
		self.___nombre = aNombre

	def getLocation(self) -> Locacion:
		return self.___location

	def setLocation(self, aLocation : Locacion) -> None:
		self.___location = aLocation

	def __init__(self):
		self.___nombre : String = None
		self.___location : Locacion = None
		self._atLocation_location___Location_ : Biosenosis = None
		self._laLocacion : Locacion = None
		"""# @AssociationMultiplicity 1
		# @AssociationKind Aggregation"""

